examples/composing-sandbox-nw-policies/README.md

```diff
@@ -0,0 +1,86 @@
+*  **Using a tool like KRO:** [KRO (Kubernetes Resource Orchestrator)](https://kro.run/docs/overview) allows you to define composite resources declaratively using a `ResourceGraphDefinition` (RGD). This creates a server side CRD that can be consumed by users.
+* **Using Helm:** Similar to KRO, we can use Helm to create a client side composition of resources and use the helm-cli to manage it.
+
+This example demonstrates the second approach, using KRO to define a new `AgenticSandbox` CRD that encapsulates a `Sandbox`, a `NetworkPolicy`, and a `Service`.
+
+The `Service` created here is distinct from the one created by the `agent-sandbox` controller. This layering allows for more complex networking configurations and showcases the value of composing resources.
+
+## Brief introduction to KRO
+
+KRO is a powerful tool for defining and managing composite resources in Kubernetes. At its core, KRO introduces the `ResourceGraphDefinition` (RGD) custom resource. An RGD allows you to declaratively specify a graph of Kubernetes resources that should be created, updated, and deleted together as a single unit. When an RGD is applied to a cluster, KRO automatically generates a new Custom Resource Definition (CRD) based on the RGD's schema. Users can then interact with this new, higher-level CRD, and KRO will handle the orchestration of the underlying resources defined in the RGD. This enables administrators to define complex, opinionated abstractions that users can consume easily, without needing to manage the individual components.
+
+## Example: Using KRO to create an `AgenticSandbox`
+
+### Install KRO
+
+First, install KRO on your cluster:
+```
+dev/tools/install-kro
+```
+
+### Create RGD
+
+Install the `ResourceGraphDefinition` (RGD) which defines our new `AgenticSandbox` CRD
+
+```
+kubectl apply -f rgd.yaml
+```
+
+Validate that the RGD is installed correctly:
+
+```
+$ kubectl get rgd
+NAME              APIVERSION   KIND             STATE    AGE
+agentic-sandbox   v1alpha1     AgenticSandbox   Active   6m38s
+```
+
+Validate that the new `AgenticSandbox` CRD is installed correctly:
+```
+$ kubectl get crd
+NAME                                      CREATED AT
+agenticsandboxes.custom.agents.x-k8s.io   2025-09-20T05:03:49Z  # << THIS
+resourcegraphdefinitions.kro.run          2025-09-20T04:35:37Z
+sandboxes.agents.x-k8s.io                 2025-09-19T22:40:05Z
+```
+
+### Create an `AgenticSandbox`
+
+You can now create an `AgenticSandbox` resource, which will create the underlying `Sandbox`, `NetworkPolicy`, and `Service` resources.
+
+Here is an example `AgenticSandbox` resource:
+
+```yaml
+apiVersion: custom.agents.x-k8s.io/v1alpha1
+kind: AgenticSandbox
+metadata:
+  name: demo
+spec:
+  image: nginx
+  service:
+    port: 80
+```
+
+Apply it to the cluster:
+```
+kubectl apply -f instance.yaml
+```
+
+You can then check the status of the created resources:
+
+```
+kubectl get agenticsandboxes
+kubectl get agenticsandboxes demo -o yaml
+```
```

Comment:
Can we showcase how users can interact/use agenticsandboxes? To answer the "so what" question

Reply:
took a stab at it. PTAL.